src/Build/TelemetryInfra/InternalTelemetryConsumingLogger.cs

```diff
@@ -75,6 +82,22 @@ private void FlushDataIntoConsoleIfRequested()
         Console.WriteLine("==========================================");
     }
 
+    private void FlushDataIntoJsonFileIfRequested()
+    {
+        const string jsonFileNameVariable = "MSBUILDNODETELEMETRYFILENAME";
```

Comment:
do you need to read this var all the time on method run? can it be moved to a global var?